src/main/java/apoc/metrics/Metrics.java

```diff
@@ -162,10 +166,18 @@ public boolean test(LoadCsv.CSVResult o) {
                     "https://neo4j.com/docs/operations-manual/current/monitoring/metrics/expose/#metrics-csv");
         }
 
-        String url = new File(metricsDir, metricName + ".csv").toURI().toString();
```

Comment:
What is the difference that makes it not fail with the new, more restrictive config setting?

Reply:
The old way leverages directly the `LoadCsv#csv` method which uses `FileUtils#readerFor` and it internally uses:
`FileUtils#changeFileUrlIfImportDirectoryConstrained` which is the cause of the issue; so I created the `LoadCsv#streamCsv` method that thakes a reader in input.